sound/soc/sof/sof-audio.h

```diff
@@ -399,6 +405,18 @@ struct snd_sof_widget {
 	u32 num_sink_pins;
 	u32 num_source_pins;
 
+	/* The sink/source pin binding array, it takes the form of
+	 * [widget_name_connected_to_pin0, widget_name_connected_to_pin1, ...],
+	 * with the index as the queue ID.
+	 *
+	 * The array is used for special pin binding. Note that even if there
+	 * is only one sink/source pin requires special pin binding, pin binding
+	 * should be defined for all sink/source pins in topology, for pin(s) that
+	 * are not used, give the value "NotConnected".
```

Comment:
You need to add a statement for the generic case, e.g. the pin binding shall be NULL.